compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -150,7 +150,11 @@ struct ConvertSharedMemAllocOp : public OpRewritePattern<memref::AllocOp> {
       if (auto shapeType = llvm::dyn_cast<ShapedType>(elType))
         alignement =
             shapeType.getNumElements() * shapeType.getElementTypeBitWidth() / 8;
-      else
+      else if (elType.isIndex()) {
+        auto mod = allocOp->getParentOfType<ModuleOp>();
+        LowerToLLVMOptions options(mod.getContext(), DataLayout(mod));
+        alignement = options.getIndexBitwidth() / 8;
```

Comment:
@antiagainst Should we add a functor as well here for the index bitwidth like what you suggested in https://github.com/openxla/iree/pull/14001 ?

Reply:
No need. This is for LLVMGPU itself and not shared with SPIR-V. (Passes shared with SPIR-V are under `Codegen/Common/GPU`.)